drivers/pulse_counter/pulse_counter_saul.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2017 UC Berkeley
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_pulse_counter
+ * @{
+ *
+ * @file
+ * @brief       PULSE_COUNTER adaption to the RIOT actuator/sensor interface
+ *
+ * @author      Hyung-Sin Kim <hs.kim@cs.berkeley.edu>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "saul.h"
+#include "pulse_counter.h"
+
+static int read_pulse_counter(const void *dev, phydat_t *res)
+{
+    res->val[0] = pulse_counter_read((const pulse_counter_t*)dev);
+    memset(&(res->val[1]), 0, 2 * sizeof(int16_t));
```

Comment:
No need to touch the unused values. The return value 1 tells the caller that the other values are unused and garbage.

Reply:
fixed, thanks!